chunk/schema.go

```diff
@@ -602,50 +682,71 @@ func decodeBase64Value(bs []byte) (model.LabelValue, error) {
 	return model.LabelValue(decoded), nil
 }
 
-func parseRangeValue(v []byte) (model.LabelValue, string, error) {
+func parseRangeValue(rangeValue []byte, value []byte) (Chunk, model.LabelValue, error) {
 	components := make([][]byte, 0, 5)
 	i, j := 0, 0
-	for j < len(v) {
-		if v[j] != 0 {
+	for j < len(rangeValue) {
+		if rangeValue[j] != 0 {
 			j++
 			continue
 		}
 
-		components = append(components, v[i:j])
+		components = append(components, rangeValue[i:j])
 		j++
 		i = j
 	}
 
 	switch {
 	case len(components) < 3:
-		return "", "", fmt.Errorf("invalid range value: %x", v)
+		return Chunk{}, "", fmt.Errorf("invalid range value: %x", rangeValue)
 
 	// v1 & v2 schema had three components - label name, label value and chunk ID.
 	// No version number.
 	case len(components) == 3:
-		return model.LabelValue(components[1]), string(components[2]), nil
+		chunk := Chunk{ID: string(components[2])}
+
+		if value != nil {
+			if err := json.Unmarshal(value, &chunk); err != nil {
+				return Chunk{}, "", err
+			}
+			chunk.metadataInIndex = true
+		}
+
+		return chunk, model.LabelValue(components[1]), nil
 
 	// v3 schema had four components - label name, label value, chunk ID and version.
 	// "version" is 1 and label value is base64 encoded.
 	case bytes.Equal(components[3], rangeKeyV1):
-		value, err := decodeBase64Value(components[1])
-		return value, string(components[2]), err
+		chunk := Chunk{ID: string(components[2])}
+		labelValue, err := decodeBase64Value(components[1])
+		return chunk, labelValue, err
 
 	// v4 schema wrote v3 range keys and a new range key - version 2,
 	// with four components - <empty>, <empty>, chunk ID and version.
 	case bytes.Equal(components[3], rangeKeyV2):
-		return "", string(components[2]), nil
+		chunk := Chunk{ID: string(components[2])}
+		return chunk, model.LabelValue(""), nil
 
 	// v5 schema version 3 range key is chunk end time, <empty>, chunk ID, version
 	case bytes.Equal(components[3], rangeKeyV3):
-		return "", string(components[2]), nil
+		chunk := Chunk{ID: string(components[2])}
+		return chunk, model.LabelValue(""), nil
 
 	// v5 schema version 4 range key is chunk end time, label value, chunk ID, version
 	case bytes.Equal(components[3], rangeKeyV4):
-		value, err := decodeBase64Value(components[1])
-		return value, string(components[2]), err
+		chunk := Chunk{ID: string(components[2])}
+		labelValue, err := decodeBase64Value(components[1])
+		return chunk, labelValue, err
+
+	// v6 schema added version 5 range keys, which have the label value written in
```

Comment:
Clarify that there's still 4 components, but the second one is empty? Like in the comment for v3.